synthetic_data_generation.py

```diff
@@ -111,12 +113,40 @@ def noise_generator(x):
         axs[i].axvline(x=true_freq, color='k', linestyle='--', alpha=0.4) #plotting vertical lines at the true frequencies            
 
 fig.tight_layout()
+
+
 # %%
 # now that we better understand how to use frequency estimation tools
 
 # lets try to simulate a signal with powerline noise.
 
 # create a signal that has powerline noise at 60Hz and at 6 harmonics (120, 180, 240, ...)
 # plot 1: plot the signal
+FREQUENCY = 60
+SAMPLE_RATE = 2000
+DURATION_S = 10
+N_HAR = 6
+t, sinewave = sine_wave_generator(DURATION_S, SAMPLE_RATE, FREQUENCY, n_harmonics=N_HAR) 
+
+fig, (ax1, ax2) = plt.subplots(2, 1, figsize=(10,4), dpi=200)
+ax1.plot(t, sinewave, 'k')
+ax1.spines['top'].set_visible(False)
+ax1.spines['right'].set_visible(False)
+ax1.set_xlim([0,1])
+ax1.set_ylabel("Power")
+ax1.set_xlabel("Time (s)")
+ax1.set_title("Time Domain")
 # plot 2: compute the FFT using appropriate parameters for the power spectral density estimate (using signal.welch) and plot the spectra
-
+#Use sampling rate/NFFT = frequency resolution (0.1) to find ideal NFFT value
+#Use .2 (NFFT) = NOVERLAP
+
+f, pxx = signal.welch(sinewave, nperseg = 20000, nfft = 20000, noverlap = 4000, fs = SAMPLE_RATE)
```

Comment:
Rather than directly hard-coding the values of `nperseg`, `nfft`, and `noverlap`, I would parameterize these. We always want `nperseg` == `nfft` so we should just set one and then make sure the other references that value. e.g.,

```
NFFT = 20000
NOVERLAP = 4000
NPERSEG = NFFT
f, pxx = signal.welch(sinewave, nperseg=NPERSEG, nfft=NFFT, noverlap=NOVERLAP, fs = SAMPLE_RATE)
```

This reduces the likelihood of a bug slipping in by incorporating constraints on the values the parameters can take on implicitly through defining our parameters as Python variables